metsim/disaggregate.py

```diff
@@ -7,107 +7,172 @@
 import itertools
 import scipy
 
-from metsim.configuration import PARAMS as params
-from metsim.configuration import CONSTS as consts
+import metsim.constants as cnst
+from metsim.physics import svp
 
-def disaggregate(df_daily, solar_geom):
+def disaggregate(df_daily, params, solar_geom):
     """
     TODO
     """
-    stop = params['stop'] + pd.Timedelta('1 days')     
+    stop = params['stop'] + pd.Timedelta('1 days')
     dates_disagg = pd.date_range(params['start'], stop, freq=params['time_step']+'T')
     df_disagg = pd.DataFrame(index=dates_disagg)
-    df_disagg['shortwave'] = (shortwave(df_daily['swrad'],
+    n_days = len(df_daily['day_of_year'])
+    n_disagg = len(df_disagg)
+    ts = float(params['time_step'])
+
+    df_disagg['shortwave'] = shortwave(df_daily['swrad'],
                                        df_daily['dayl'],
                                        df_daily['day_of_year'],
-                                       solar_geom['tiny_rad_fract']))
-    df_disagg['temp'] = temp(df_daily, df_disagg)
-    df_disagg['precip'] = precip(df_daily['precip'])
-    df_disagg['longwave'] = longwave(df_daily['lwrad'])
-    df_disagg['wind'] = wind(df_daily['wind'])
+                                       solar_geom['tiny_rad_fract'],
+                                       params)
+
+    t_Tmin, t_Tmax = set_min_max_hour(df_disagg['shortwave'], n_days, ts)
+    df_disagg['temp'] = temp(df_daily, df_disagg, t_Tmin, t_Tmax, ts)
+    
+    df_disagg['vapor_pressure'] = vapor_pressure(df_daily['vapor_pressure'],
+                                                 df_disagg['temp'],
+                                                 t_Tmin, n_disagg, ts)
+
+    df_disagg['rel_humid'] = relative_humidity(df_disagg['vapor_pressure'],
+                                               df_disagg['temp'])
+
+    df_disagg['longwave'], df_disagg['tskc'] = longwave(df_disagg['temp'],
+                                                        df_disagg['vapor_pressure'],
+                                                        df_daily['tskc'])
+
+    df_disagg['precip'] = precip(df_daily['precip'], ts)
+    df_disagg['wind'] = wind(df_daily['wind'], ts)
+
     return df_disagg.fillna(method='ffill')
 
 
-def set_min_max_hour(disagg_rad, n_days):
+def set_min_max_hour(disagg_rad, n_days, ts):
     """
     TODO
     """
-    ts = float(params['time_step'])
     rad_mask = 1*(disagg_rad > 0)
     diff_mask = np.diff(rad_mask)
-    rise_times = np.where(diff_mask>0)[0] * ts 
+    rise_times = np.where(diff_mask>0)[0] * ts
     set_times = np.where(diff_mask<0)[0] * ts
-    t_Tmax = 0.67 * (set_times - rise_times) + rise_times
-    t_Tmin = rise_times - float(params['time_step'])
-    return t_Tmin, t_Tmax
+    t_t_max = 0.67 * (set_times - rise_times) + rise_times
```

Comment:
Pull out magic number (`0.67`) as much as possible and replace them by a variable. In this case, the value would be added to `constants.py` with a brief description